### ISSUE TITLE
Help required with the SUGroup function

### ISSUE THREAD

**Issue by RJoshlan** [SantiagoEG/FCBF_module#2]
Hi, i wanted help with implementing the SUGroup method in java if your familiar with java. at the moment i have implemented the basic in java but i m not sure where i should call the symmetric uncertain method. 

This is your code in python:
`
def suGroup(x, n):
    m = x.shape[0]
    x = np.reshape(x, (n,m/n)).T
    m = x.shape[1]
    SU_matrix = np.zeros(shape = (m,m))
    for j in range(m-1):
        x2 = x[:,j+1::]
        y = x[:,j]
        temp = np.apply_along_axis(symmetricalUncertain, 0, x2, y)
        for k in range(temp.shape[0]):
            SU_matrix[j,j+1::] = temp
            SU_matrix[j+1::,j] = temp
    
    return 1/float(m-1)*np.sum(SU_matrix, axis = 1)`

This my code in java (unfinished- doesn't include the SU function)

`public static final double correlation(double X[], double y[]) {

		int n =  X.length; 
		int i;
		double sum_of_x = 0.0, sum_of_y = 0.0, sum_of_sqrArr_X = 0.0, sum_of_sqrArr_y = 0.0, sum_of_XY = 0.0;
		double corr;		
		if (n <= 1) {
			return 1.0;
		}
		for (i = 0; i < n; i++) {
			sum_of_x += X[i];
			sum_of_y += y[i];
		}
		sum_of_x /= (double) n;
		sum_of_y /= (double) n;
		for (i = 0; i < n; i++) {
			sum_of_sqrArr_X += (X[i] - sum_of_x) * (X[i] - sum_of_x);
			sum_of_sqrArr_y += (y[i] - sum_of_y) * (y[i] - sum_of_y);
			sum_of_XY += (X[i] - sum_of_x) * (y[i] - sum_of_y);
		}
		if (sum_of_sqrArr_X * sum_of_sqrArr_y == 0.0) {
			corr = 1.0;
		} else {
			corr = sum_of_XY / Math.sqrt(Math.abs(sum_of_sqrArr_X * sum_of_sqrArr_y));
		}

		return corr;
	}`

**Comment by SantiagoEG** [comment]
Dear RJoshlan,

sorry for my delay in replying, I've been very busy these days.
Let my explain the line in which I call symmetricalUncertain function.

*temp = np.apply_along_axis(symmetricalUncertain, 0, x2, y) *applies the
function *symmetricalUncertain* along axis *0* to the array *x2* and using
*y* as second argument for the function. More info here:
https://numpy.org/doc/stable/reference/generated/numpy.apply_along_axis.html

So, basically, I am preventing using a for loop by this method. I do not
know if this is possible in Java, since I do not have a deep knowledge
about it, but you could replace this line by a for loop.

I hope this could help you, and I apoligize because I am not an expert on
Java.

Best,
Santiago

**Comment by RJoshlan** [comment]
Thank you. The issue is sorted. It was just the for loop i was confused about in the su function but rest is good. thanks for replying
